var/spack/repos/builtin/packages/aocl-sparse/package.py

```diff
@@ -0,0 +1,116 @@
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+import inspect
+
+
+class AoclSparse(CMakePackage):
+    """AOCL-Sparse is a library that contains basic linear algebra subroutines
+    for sparse matrices and vectors optimized for AMD EPYC family of processors.
+    It is designed to be used with C and C++. Current functionality of sparse
+    library supports SPMV function with CSR and ELLPACK formats."""
+
+    homepage = "https://developer.amd.com/amd-aocl/aocl-sparse/"
+    url = "https://github.com/amd/aocl-sparse/archive/2.2.tar.gz"
+    git = "https://github.com/amd/aocl-sparse.git"
+
+    maintainers = ['amd-toolchain-support']
+
+    version('2.2',  sha256='33c2ed6622cda61d2613ee63ff12c116a6cd209c62e54307b8fde986cd65f664')
+
+    conflicts("%gcc@:9.1.999", msg="Minimum required GCC version is 9.2.0")
+
+    variant('shared', default=True,
+            description='Build shared library')
+    variant('ilp64', default=False,
+            description='Build with ILP64 support')
+    variant('test', default=False,
+            description='Build with test application with aocl-sparse')
+    variant('build_type', default='Release',
+            description='CMake build type',
+            values=('Debug', 'Release'))
+
+    depends_on('boost', when='+test')
+
+    @property
+    def build_directory(self):
+        """Returns the directory to use when building the package
+
+        :return: directory where to build the package
+        """
+
+        builddir = self.stage.source_path
+
+        if self.spec.variants['build_type'].value == 'Debug':
+            builddir = join_path(self.stage.source_path, 'build', 'debug')
+        else:
+            builddir = join_path(self.stage.source_path, 'build', 'release')
+
+        mkdirp(builddir)
+        return builddir
+
+    def cmake(self, spec, prefix):
+        """Runs ``cmake`` in the build directory"""
+        args = [
+            "../..",
+            "-DCMAKE_INSTALL_PREFIX:PATH={0}".format(spec.prefix),
+            "-DCMAKE_CXX_COMPILER={0}".format(os.path.basename(spack_cxx))
+        ]
+
+        if spec.variants['build_type'].value == 'Debug':
+            args.append("-DCMAKE_BUILD_TYPE=Debug")
+        else:
+            args.append("-DCMAKE_BUILD_TYPE=Release")
+
+        args.extend([
+            "-DBUILD_SHARED_LIBS:BOOL=%s" % (
+                'ON' if '+shared' in spec else 'OFF'
+            )
+        ])
+
+        args.extend([
+            "-DBUILD_ILP64:BOOL=%s" % (
+                'ON' if '+ilp64' in spec else 'OFF'
+            )
+        ])
+
+        args.extend([
+            "-DBUILD_CLIENTS_BENCHMARKS:BOOL=%s" % (
+                'ON' if '+test' in spec else 'OFF'
+            )
+        ])
+
+        with working_dir(self.build_directory, create=True):
+            inspect.getmodule(self).cmake(*args)
+
+    # Check that self.prefix is there after installation
+    @run_after('build')
```

Comment:
Curious.  Why check after the `build` phase?  In general, tests are run after the `install` phase.

Reply:
when user enters +test option, make install don’t copy “tests/staging/aoclsparse-bench” file required for unit testing along with library deliverables (i.e, include and lib directory), so we have pre-empt this activity.